src/Framework/EvaluationContext/EvaluationContextFactory.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//-----------------------------------------------------------------------
+// </copyright>
+
+namespace Microsoft.Build.Framework.EvaluationContext
+{
+    /// <summary>
+    /// Factory for <see cref="EvaluationContext"/>
+    /// </summary>
+    public abstract class EvaluationContextFactory
```

Comment:
Do we need a factory?  Can there be a static method on `EvaluationContext` so the pattern is aligned with stuff like `ProjectRootElement.Create()`